Screens/SignUpScreen.js

```diff
@@ -1,28 +1,36 @@
 import React, { useState } from "react";
-import {
-  TouchableOpacity,
-  View,
-  Text,
-  TextInput,
-  Button,
-  StyleSheet,
-} from "react-native";
+import { View, StyleSheet, TouchableOpacity } from "react-native";
+import { Text } from "react-native-paper";
+import Background from "../components/Background";
+import Logo from "../components/Logo";
+import Header from "../components/Header";
+import Button from "../components/Button";
+import TextInput from "../components/TextInput";
+import BackButton from "../components/BackButton";
+import { theme } from "../core/theme";
+import { usernameValidator } from "../utils/UsernameValidator";
+import { passwordValidator } from "../utils/PasswordValidator";
 
-import axios from "axios";
 import AsyncStorage from "@react-native-async-storage/async-storage";
 import RestApiService from "../services/RestApiService";
 
 const SignUpScreen = ({ navigation }) => {
-  const [username, setUsername] = useState("");
-  const [email, setEmail] = useState("");
-  const [password, setPassword] = useState("");
+  const [username, setUsername] = useState({ value: "", error: "" });
+  const [password, setPassword] = useState({ value: "", error: "" });
 
-  const handleLogin = async () => {
-    // Implement your login logic here
+  const handleSignUp = async () => {
```

Comment:
Almost the same functionality as in handleLogin. Maybe we can extract common logic?